site/hooks/useBalance.js

```diff
@@ -0,0 +1,35 @@
+import { useContext } from 'react'
+import { useWeb3React } from '@web3-react/core'
+import useSWR from 'swr'
+import PureContext from '../components/context/Pure'
+
+export const useBalance = function ({ symbol }) {
+  const { active, account: userAccount, library } = useWeb3React()
+  const { erc20 } = useContext(PureContext)
+
+  const { data, mutate, error } = useSWR(
+    [`${symbol}-Balance-${userAccount}`, userAccount, erc20],
+    async function (_, account, erc20Instancer) {
+      if (!active) {
+        return Promise.resolve(null)
+      }
+      const erc20Service = erc20Instancer?.(account)
+      switch (symbol) {
+        case 'ETH':
+          return library.eth.getBalance(account, 'latest')
+        case 'WETH':
+          return erc20Service.wrappedEtherBalanceOf()
```

Comment:
Will this fail when `erc20Service` is `null`?

Reply:
in the previous lines I'm checking for `isActive`, so at this point there should be an account defined. Considering that `erc20Instancer` is just a function defined in the context returning an object, I'd hope this to never fail (I even think I can remove the `?.` in the `erc20Instancer` 🤔 I think I added it before the `isActive` check)